src/expr/impl/src/scalar/jsonb_concat.rs

```diff
@@ -59,43 +59,35 @@ use serde_json::{json, Value};
 /// ```
 #[function("jsonb_cat(jsonb, jsonb) -> jsonb")]
 pub fn jsonb_cat(left: JsonbRef<'_>, right: JsonbRef<'_>) -> JsonbVal {
-    let left_val = left.value().clone();
-    let right_val = right.value().clone();
-    match (left_val, right_val) {
+    match (left.into(), right.into()) {
         // left and right are object based.
         // This would have left:{'a':1}, right:{'b':2} -> {'a':1,'b':2}
-        (Value::Object(mut left_map), Value::Object(right_map)) => {
-            left_map.extend(right_map);
-            JsonbVal::from(Value::Object(left_map))
+        (ValueRef::Object(left), ValueRef::Object(right)) => {
+            JsonbVal::from(Value::object(left.iter().chain(right.iter())))
```

Comment:
Let's also document the `last-win` behavior on duplicate keys in jsonbb.

Reply:
Another comment regarding jsonbb object key deduplication: seems the deduplication only handles entries, and the actual data of key and value removed due to duplication are still in the buffer but inaccessible. Is this understanding correct?

Reply:
Btw, this reminds me a flaw in jsonbb. The overwritten key-value pairs won't be accessed, but they **still exist** in the jsonbb encoding. This is a compromise for building performance. When building the JSON, it only writes sequentially to the buffer, avoiding additional memory allocations and movements. But this flaw could lead to space waste and potential infinite expansion. 🤪

Reply:
Yes, I was about to mention that.

Reply:
I should remove the data along with entries. But the price is additional memory movement. Seems there's no better solution. 🤔

Reply:
We can iterate on the implementation after this PR, without hurting backward compatibility.